src/parser/mod.rs

```diff
@@ -159,15 +217,27 @@ impl Parser {
                 }
             }
 
-            return Ok(Type::Intersection(types));
+            return Ok(Type::Intersection(Box::new(types)));
         }
 
+        Self::parse_type(id)
+    }
+
+    fn parse_type(id: ByteString) -> ParseResult<Type> {
```

Comment:
Could we maybe move this to a regular function instead of a static method on the struct?